pkg/controller/containerjfr/containerjfr_controller.go

```diff
@@ -277,6 +277,25 @@ func (r *ReconcileContainerJFR) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
+	// Check that secrets mounted in /truststore coincide with CRD
+	err = r.Client.Get(context.Background(), types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}, deployment)
+	if err == nil {
+		deploymentMounts := deployment.Spec.Template.Spec.Containers[0].VolumeMounts
+		expectedMounts := resources.NewDeploymentForCR(instance, serviceSpecs, tlsConfig).Spec.Template.Spec.Containers[0].VolumeMounts
+		if !equals(deploymentMounts, expectedMounts) {
+			reqLogger.Info("cert secrets mounted do not conicide with those specified in CRD, modifying deployment")
+			// Modify deployment
+			deployment := resources.NewDeploymentForCR(instance, serviceSpecs, tlsConfig)
```

Comment:
I would just do `deployment.Spec.Template.Spec.Containers[0].VolumeMounts = expectedMounts` here instead of replacing the whole deployment.